discovery/discovery.py

```diff
@@ -20,39 +23,84 @@
 from utils.decorators.persist_execution import persistence
 from utils.visualizer import Visualizer
-from utils.metadata import Metadata, ColMetadata
 
 
 class Discovery:
     def __init__(self, discovery_config: dict):
         self.config = discovery_config
         self.file_handler = FileHandler()
-        self.file_metadata = []
+        self.dataframe_file_metadata_pairs = []
         self.visualiser = Visualizer()
 
     def create_visual(self, pathname):
         """
         Build a visual based on stored metadata
         """
-        self.visualiser.draw(self.file_metadata, pathname)
+
+        # TODO: find a fancy python way to do this
+        metadata = []
+        for dataframe, metadatum in self.dataframe_file_metadata_pairs:
+            metadata.append(metadatum)
+        self.visualiser.draw(metadata, pathname)
 
     def reconstruct_metadata(self):
         """
         Builds metadata based on the currently loaded files
         """
         file_metadata = []
-        for path, dataframe in self.file_handler.loaded_files.items():
-            col_meta = []
-            for col_name in dataframe.columns:
-                average, col_min, col_max = None, None, None
-                if is_numeric_dtype(dataframe[col_name]):
-                    average = dataframe[col_name].mean()
-                    col_min = dataframe[col_name].min()
-                    col_max = dataframe[col_name].max()
-                col_meta.append(ColMetadata(col_name, dataframe[col_name].dtype, average, col_min, col_max))
-
-            file_metadata.append(Metadata(path, col_meta))
-        self.file_metadata = file_metadata
+        for path, file_descriptor in self.file_handler.loaded_files.items():
+            metadatum = self._construct_metadatum(file_descriptor)
+            file_metadata.append((file_descriptor["dataframe"], metadatum))
+        self.dataframe_file_metadata_pairs = file_metadata
+
+    def _construct_metadatum(self, file_descriptor):
+        metadatum = Metadata(file_descriptor["file_path"], file_descriptor["extension"],
+                             file_descriptor["size"], file_descriptor["file_hash"])
+        col_meta = []
+        dataframe = file_descriptor["dataframe"]
+
+        for col_name in dataframe.columns:
+            column_data = self._construct_column(dataframe[col_name])
+            col_meta.append(column_data)
+        metadatum.columns = col_meta
+        return metadatum
+
+    def construct_relationships(self):
+        for reference_dataframe, reference_metadatum in self.dataframe_file_metadata_pairs:
+            for subject_dataframe, subject_metadatum in self.dataframe_file_metadata_pairs:
+                if reference_metadatum.hash == subject_metadatum.hash:
+                    continue
+                self._match_metadata(reference_dataframe, reference_metadatum, subject_dataframe, subject_metadatum)
+                print(reference_metadatum.file_path+" "+subject_metadatum.file_path)
```

Comment:
user log calls instead of print statements if its not debug (it lets us build config files which handle output)